Content.Shared/Blocking/BlockingSystem.cs

```diff
@@ -128,12 +132,42 @@ public bool StartBlocking(EntityUid item, BlockingComponent component, EntityUid
 
         if (component.BlockingToggleAction != null)
         {
+            //Don't allow someone to block if they're in a container.
             if (_containerSystem.IsEntityInContainer(user) || !_mapManager.TryFindGridAt(xform.MapPosition, out var grid))
             {
                 CantBlockError(user);
                 return false;
             }
 
+            //Don't allow someone to block if someone else is on the same tile.
+            var playerTileRef = xform.Coordinates.GetTileRef();
+            if (playerTileRef != null)
+            {
+                var intersecting = _lookup.GetEntitiesIntersecting(playerTileRef.Value);
+                var query = GetEntityQuery<MobStateComponent>();
+
+                foreach (var uid in intersecting)
+                {
+                    if (query.HasComponent(uid) && uid != user)
+                    {
+                        TooCloseError(user);
+                        return false;
+                    }
+                }
+            }
+
+            //Don't allow someone to block if they're in a doorway
+            var intersectingAnchoredEntities = _lookup.GetEntitiesIntersecting(user, LookupFlags.Anchored);
+            foreach (var entity in intersectingAnchoredEntities)
+            {
+                if (HasComp<DoorComponent>(entity))
+                {
+                    TooCloseError(user);
+                    return false;
+                }
+            }
```

Comment:
Just get intersecting first and then iterate once and check mobstate / door as this will do the query twice. Also check uid != user first as that's faster.